test/cql-pytest/test_compaction_strategy_timestamp_resolution.py

```diff
@@ -0,0 +1,21 @@
+# Copyright 2023-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+
+#############################################################################
+# Tests for TWCS timestamp resolution validation
+#############################################################################
+
+import pytest
+from util import new_test_table
+from cassandra.protocol import ConfigurationException
+
+@pytest.fixture(scope="module")
+def table1(cql, test_keyspace):
+    with new_test_table(cql, test_keyspace, "a int PRIMARY KEY, b int", "WITH compaction = { 'class' : 'TimeWindowCompactionStrategy' }") as table:
+        yield table
+
+def test_timestamp_resolution_options(cql, table1):
+    timestamp_resolutions = ["MICROSECONDS", "MILLISECONDS", "SECONDS", "MINUTES", "HOURS", "DAYS"]
+    for tr in timestamp_resolutions:
+        cql.execute(f"ALTER TABLE {table1} WITH compaction = {{ 'class' : 'TimeWindowCompactionStrategy', 'timestamp_resolution' : '{tr}' }}")
```

Comment:
please just add at least one error case, just to make sure that ALTER TABLE doesn't simply ignore its parameter.

Since this test modifies the table, I suggest that you don't use a fixture - and just use `with new_test_table` directly in the test - the test creates its own table, does a bunch of modifications to its compaction strategy.

Then, you'll be left with a single 5-line test in a new file, maybe you can put it in an existing file instead of creating a new file? Isn't test/cql-pytest/test_compaction_strategy_validation.py a good place? It's actually where we already have a test that found this problem originally.

Reply:
> Isn't test/cql-pytest/test_compaction_strategy_validation.py a good place?

Agree, added two test cases to the existing `test_compaction_strategy_validation.py` file: one to test table creating with TWCS time resolution option and another one for altering table.

> please just add at least one error case, just to make sure that ALTER TABLE doesn't simply ignore its parameter.

Added error cases for both tests.